libminifi/include/core/flow/StructuredConfiguration.h

```diff
@@ -34,6 +34,8 @@
 #include "utils/file/FileSystem.h"
 #include "core/flow/Node.h"
 #include "FlowSchema.h"
+#include "rapidjson/document.h"
+#include "yaml-cpp/yaml.h"
```

Comment:
although it would be nice not to have this this dependency here, I understand that it would require extra indirections/tests which would bloat this PR, could we create a ticket to tuck the clone/modification/save implementations away like we do with flow::Node?

Reply:
I've created https://issues.apache.org/jira/browse/MINIFICPP-2281 for this